### PR TITLE
DLS-1854 for comprehension replaced with multiple folds and flatMap

### PR DESCRIPTION
**Pull request by rajanimohan22**
'for' comprehension replaced with multiple folds and flatMap

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala [hunk 1]
```diff
@@ -80,44 +79,42 @@ class ReminderController @Inject() (val helpToSaveReminderService: HelpToSaveRem
...
-              })
-          }
-
-        }
-    )
+                },
+                emailRetrieved ⇒
+                  emailRetrieved match {
+                    case Some(email) ⇒
+                      if (email.length > 0)
```
Comment (lutnos) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: You can do this more tidily using a guard, so say:

      case Some(email) if !email.isEmpty ⇒
 
and later

      case Some(_) =>
                      logger.warn(s"Empty email retrieved for user: ${userInfo.nino}")

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala [hunk 2]
```diff
@@ -80,44 +79,42 @@ class ReminderController @Inject() (val helpToSaveReminderService: HelpToSaveRem
...
+                        helpToSaveReminderService.updateHtsUser(HtsUser(Nino(htsContext.nino), email, userInfo.forename, true, daysToReceive = DateToDaysMapper.d2dMapper.getOrElse(success.reminderFrequency, Seq(1))))
```
Comment (lutnos) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: This line is too long, my personal preference would be to introduce a val for daysToReceive parameter

Comment (lutnos) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: That still comes out at more than 160 character for the line, I actually meant something like:

                      val daysToRetrieve = DateToDaysMapper.d2dMapper.getOrElse(success.reminderFrequency, Seq())
                      helpToSaveReminderService.updateHtsUser(HtsUser(Nino(htsContext.nino), email, userInfo.forename, true, daysToRetrieve))

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala (hunk 1 above)
Comment (lutnos) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: Great, thanks

File: app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala (hunk 2 above)
Comment (rajanimohan22) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala]: Ok, let me do this

**Pull request merged**